### ISSUE TITLE
Client Specific Overrides are generated in /tmp

### ISSUE THREAD

**Issue by floek** [opnsense/core#5562]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [X] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [X] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Describe the bug**
Hi,

I'm on 21.7.7. My Client Specific Overrides are generated in /tmp according to my logs:
```
client config created @ /tmp/openvpn_cc_1e346b7b87c0ce6e76673802950277cb.tmp
```

According to the generated config, it should be created under:
```
client-config-dir /var/etc/openvpn-csc/2
```

**To Reproduce**

I have two openvpn servers, one disabled, one with a client specific override:
```xml
  <openvpn>
    <openvpn-server>
      <mode>server_tls_user</mode>
      [...]
      <disable>1</disable>
    </openvpn-server>
    <openvpn-server>
      <mode>p2p_tls</mode>
      <protocol>UDP4</protocol>
      <dev_mode>tun</dev_mode>
      <local_port>1194</local_port>
      <description>Home</description>
      <crypto>AES-256-CBC</crypto>
      <digest>SHA256</digest>
      <tunnel_network>192.168.1.0/28</tunnel_network>
      <remote_network>192.168.2.0/24</remote_network>
      <local_network>192.168.3.0/24,192.168.4.0/24</local_network>
      <dynamic_ip>yes</dynamic_ip>
      <pool_enable>yes</pool_enable>
      <verbosity_level>1</verbosity_level>
      <vpnid>2</vpnid>
      <interface>any</interface>
      <custom_options/>
      <caref>XXXXXX</caref>
      <crlref/>
      <certref>XXXXXX</certref>
      <dh_length>2048</dh_length>
      <cert_depth>1</cert_depth>
    </openvpn-server>
    <openvpn-client/>
    <openvpn-csc>
      <custom_options>iroute 192.168.20.0 255.255.255.0</custom_options>
      <common_name>user1</common_name>
      <description>User 1</description>
    </openvpn-csc>
  </openvpn>
```

When I run the script manually, everything starts working:
```bash
common_name="user1" /usr/local/etc/inc/plugins.inc.d/openvpn/ovpn_setup_cso.php server2
```

**Expected behavior**

A file in `/var/etc/openvpn-csc/2/user1` with my `iroute`.

**Describe alternatives you considered**

A clear and concise description of any alternative solutions or workaround you considered.

**Screenshots**

See XML snippet.

**Relevant log files**

```
client config created @ /tmp/openvpn_cc_1e346b7b87c0ce6e76673802950277cb.tmp
```

**Additional context**

**Environment**

Software version used and hardware type if relevant, e.g.:

OPNsense 21.7.7 (amd64, OpenSSL).
virtual Machine

**Comment by floek** [comment]
Hi,

I already dived a little bit into the sources, but found no solution :(

Best,
Floek

**Comment by AdSchellevis** [comment]
Hi Floek,

I'm also not sure to be honest, the documentation seems to suggest it will use the client config file as passed through the `--client-connect` option (https://openvpn.net/community-resources/reference-manual-for-openvpn-2-4/), when a `--client-config-dir` is also specified it doesn't state if there's a difference in behaviour.

You probably have to check the logs, there's probably a reason why it ignores the file it asked for in the first place.

Best regards,

Ad

**Comment by AdSchellevis** [comment]
OpenVPN isn't chrooted, but I wouldn't expect the file to be there after login either. If I'm not mistaken OpenVPN removes the file after usage, easy to test by writing another file into the same directory from here:

https://github.com/opnsense/core/blob/00a86f74dbbefe3abbb926d98251067df53dfe8e/src/etc/inc/plugins.inc.d/openvpn.inc#L1023-L1025

Maybe the assumption that OpenVPN didn't read the file isn't the correct one or the content is different than expected.

**Comment by floek** [comment]
Right, tested it and the file is generated just fine. I tested my setup again and: Everything is working fine. So, I think I got into debugging this without any need 😇😇. Sorry bothering you, but now I understand how this client overrides work :).
